### PR TITLE
Improved parameter handling by continuous distributions

### PR DESCRIPTION
**Pull request by supreet11agrawal**
This improves the parametric handling by different distributions.
Many distributions have specific requirements for parameters which were
not implemented in Sympy.
Also added test cases for checking if further changes don't break the
specific requirements.

#### References to other Issues or PRs

Finishes #16159 

#### Release Notes

* stats
  * Improved parameter handling by continuous distributions

### CODE REVIEW

File: sympy/stats/crv_types.py [hunk 1]
```diff
@@ -922,7 +922,7 @@ def Exponential(name, rate):
     Parameters
     ==========
 
-    rate : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
+    lambda : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
```
Comment (jksuom) [inline sympy/stats/crv_types.py]: `lambda` is a keyword. It cannot be used as an identifier.

**Comment by supreet11agrawal** [comment]
Kindly add Stats label

File: sympy/stats/crv_types.py (hunk 1 above)
Comment (supreet11agrawal) [reply sympy/stats/crv_types.py]: Actually in https://docs.sympy.org/latest/modules/stats.html?highlight=stats#module-sympy.stats.crv_types 
for every distribution, parameters are defined in the way  `Symbol : Property/ies Name`

Eg.
https://github.com/sympy/sympy/blob/d2fcf9ecdfef464b26f0e118018a7817059d22a2/sympy/stats/crv_types.py#L271-L273
https://github.com/sympy/sympy/blob/d2fcf9ecdfef464b26f0e118018a7817059d22a2/sympy/stats/crv_types.py#L357-L358
https://github.com/sympy/sympy/blob/d2fcf9ecdfef464b26f0e118018a7817059d22a2/sympy/stats/crv_types.py#L502-L503
The symbols are in accordance with the distribution function written. I have made this change in that manner only.

File: sympy/stats/crv_types.py [hunk 2]
```diff
@@ -922,7 +922,7 @@ def Exponential(name, rate):
     Parameters
     ==========
 
-    rate : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
+    l : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
```
Comment (jksuom) [inline sympy/stats/crv_types.py]: This should be the same as in the function definition of `Exponential` and in `_argnames` of `ExponentialDistribution`.

Comment (supreet11agrawal) [reply sympy/stats/crv_types.py]: I have restored it back to `rate`

File: sympy/stats/crv_types.py [hunk 3]
```diff
@@ -956,16 +956,8 @@ def Exponential(name, rate):
     >>> skewness(X)
     2
 
-    >>> X = Exponential('x', 10)
```
Comment (Upabjojr) [inline sympy/stats/crv_types.py]: why has this been removed?

**Comment by Upabjojr** [comment]
Did you change the definitions of the distributions by chance? It is usually a good idea to create short PRs, they are easier to review.

File: sympy/stats/crv_types.py (hunk 3 above)
Comment (supreet11agrawal) [reply sympy/stats/crv_types.py]: The idea was to show the precise working of the exponential distribution. Substituting a number in place of `lambda` was obvious and increased the length of docs unnecessarily. Examples are shown with the help of a general parameter `l`.
Complete properties of Exponential distribution are described in form of a general parameter `lambda`.
https://github.com/sympy/sympy/blob/f552493265c8f15695f8254eb1812873c25f33e3/sympy/stats/crv_types.py#L942-L960

**Comment by supreet11agrawal** [comment]
> Did you change the definitions of the distributions by chance? It is usually a good idea to create short PRs, they are easier to review.

No, I haven't changed the definitions of distributions. However, I removed some minor errors from some distributions. For eg in PR #16159, I changed the Cauchy distribution for which cdf was written in docs instead of pdf.
I'll keep in mind to make small PRs next time.
Thanks

File: sympy/stats/crv_types.py [hunk 4]
```diff
@@ -1353,8 +1360,8 @@ def GammaInverse(name, a, b):
     Parameters
     ==========
 
-    a : Real number, `a > 0` a shape
-    b : Real number, `b > 0` a scale
+    alpha : Real number, `\alpha > 0` a shape
```
Comment (jksuom) [inline sympy/stats/crv_types.py]: This should agree with the call parameter of GammaInverse and _argnames of GammaInverseDistribution.

Comment (supreet11agrawal) [reply sympy/stats/crv_types.py]: Ok, I'll change it

File: sympy/stats/crv_types.py [hunk 5]
```diff
@@ -936,10 +936,10 @@ def Exponential(name, rate):
     >>> from sympy.stats import variance, std, skewness
     >>> from sympy import Symbol
 
-    >>> l = Symbol("lambda", positive=True)
+    >>> rate = Symbol("lambda", positive=True)
```
Comment (jksuom) [inline sympy/stats/crv_types.py]: It was not necessary to change this name in Examples though it does no harm.

Comment (supreet11agrawal) [reply sympy/stats/crv_types.py]: Just for consistency

File: sympy/stats/crv_types.py [hunk 6]
```diff
@@ -910,7 +910,7 @@ def _moment_generating_function(self, t):
 
 def Exponential(name, rate):
     r"""
-    Create a continuous random variable with an Exponential distribution.
+    Creates a Continuous Random Variable with Exponential distribution.
```
Comment (Upabjojr) [inline sympy/stats/crv_types.py]: I don't understand why you want to change this sentence. If you use `creates` you should add the subject first, e.g. _This function creates ..._. It makes sense to start the sentence with _Create_ but not with _Creates_.

Comment (Upabjojr) [reply sympy/stats/crv_types.py]: I would leave these sentences as they are.

File: sympy/stats/crv_types.py (hunk 3 above)
Comment (Upabjojr) [reply sympy/stats/crv_types.py]: I don't see the point of removing documentation code. A few more examples are OK.

**Comment by Upabjojr** [comment]
I would suggest to close this PR and re-open a new PR with only the necessary edits. There are lots of useless changes that make it difficult to review this PR.

**Comment by supreet11agrawal** [comment]
Ok, I'll close this PR and submit PRs distribution wise. Now, what changes should I keep? @Upabjojr @jksuom

**Pull request closed without merging**